thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1333,6 +1349,8 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
         typedef CKeyLookupLocalBase PARENT;
 
         std::vector<IKeyManager *> keyManagers;
+        std::unique_ptr<std::vector<std::atomic_bool>> keyManagersCreated; // used to ensure IKeyManager * is not used before it has fully written
```

Comment:
it's not possible to size a vector of atomic's because they have no move or copy ctor.
So you can only make a const vector size, and so need to allocate if using std::vector of atomics afaics.

Reply:
ok, That's worthy of a comment, because not obvious,
In future could also use
```
std::unique_ptr<std::atomic_bool []> x;
x.reset(new std::atomic<bool>[n]);
```